wallet/createtx.go

```diff
@@ -174,27 +174,54 @@ func (w *Wallet) txToOutputs(outputs []*wire.TxOut,
 			return err
 		}
 
-		// Wrap our coins in a type that implements the SelectableCoin
-		// interface, so we can arrange them according to the selected
-		// coin selection strategy.
-		wrappedEligible := make([]Coin, len(eligible))
-		for i := range eligible {
-			wrappedEligible[i] = Coin{
-				TxOut: wire.TxOut{
-					Value:    int64(eligible[i].Amount),
-					PkScript: eligible[i].PkScript,
-				},
-				OutPoint: eligible[i].OutPoint,
+		var inputSource txauthor.InputSource
+		if len(selectedUtxos) > 0 {
+			mapEligibleToOutpoint := make(map[wire.OutPoint]wtxmgr.
+				Credit)
+
+			for _, e := range eligible {
+				mapEligibleToOutpoint[e.OutPoint] = e
+			}
+
+			var eligibleSelectedUtxo []wtxmgr.Credit
+			for _, outpoint := range selectedUtxos {
+				e, ok := mapEligibleToOutpoint[outpoint]
+
+				if !ok {
+					return fmt.Errorf(
+						"selected outpoint"+
+							"not eligible for "+
+							"spending: %v", outpoint)
+				}
+				eligibleSelectedUtxo = append(
+					eligibleSelectedUtxo, e)
 			}
-		}
-		arrangedCoins, err := coinSelectionStrategy.ArrangeCoins(
-			wrappedEligible, feeSatPerKb,
-		)
-		if err != nil {
-			return err
-		}
 
-		inputSource := makeInputSource(arrangedCoins)
+			inputSource = constantInputSource(eligibleSelectedUtxo)
+
+		} else {
+			// Wrap our coins in a type that implements the
+			// SelectableCoin interface, so we can arrange them
+			// according to the selected coin selection strategy.
+			wrappedEligible := make([]Coin, len(eligible))
+			for i := range eligible {
+				wrappedEligible[i] = Coin{
+					TxOut: wire.TxOut{
+						Value: int64(eligible[i].
+							Amount),
+						PkScript: eligible[i].PkScript,
+					},
+					OutPoint: eligible[i].OutPoint,
+				}
+			}
+
+			arrangedCoins, err := coinSelectionStrategy.
```

Comment:
Maybe we should just rename `coinSelectionStrategy` to simply `strategy` to avoid this weird folding. The type contains enough information to warrant this shorter name IMO.

Reply:
I would change the name to that. But I am curious in the case we are not able to change the name, what would be the best way to handle this situation?

Reply:
Then the way you formatted it is would have been correct. I just really personally don't like that style, so I always try to avoid it if possible. There are a couple of tricks you can use to achieve that: Shorten the name of the struct the method is called on, shorten the method name, extract the code into its own function/method to reduce indentation, reduce indentation by avoiding `else` blocks where it makes sense (e.g. `return` in the `if`, so the `else` block can continue un-indented) and so on.